cypress/integration/common/i_click.js

```diff
@@ -1,7 +1,20 @@
 import { Then } from "cypress-cucumber-preprocessor/steps";
+
+// should('not.be.disabled') is here because it looks like cypress may choke trying to click a button that has 
+// been temporarily disabled while the page is loading. This may need ultimately to be done for more or even all steps here
  
 Then(`I click {string}`, (clickable) => {
-    cy.get("[data-cy=" + clickable + "]").click();
+    cy.get("[data-cy=" + clickable + "]").should('not.be.disabled').click();
+
+    // if this results in an Oops, then retry the click
+    // we currently believe this to be caused by the CSRF problem, which we intend to fix fully in future
+    // and this retry step will be able to be removed from the tests
+    cy.document().then(docStr => {
```

Comment:
OK, have chained this now, and logged an Oops command. After months of doing this, I've never seen it fail the second time around.  Which is not to say it never would. But I'm inclined for simplicity to leave this as it is now rather than do multiple retries, especially as effort may be better spend fixing the actual issue with the Oops page?

Reply:
OK, interesting that it only fails once, though.